### PR TITLE
Use react-listener-provider if available

### PR DESCRIPTION
**Pull request by jnsdls**
As mentioned in #1 I went ahead and moved the MouseMoveProvider idea into its own package [react-listener-provider](https://github.com/jnsdls/react-listener-provider).

In this PR I'm using that to optionally provide global event listeners. (check `/demo`)

We should probably update the docs to reflect the ability to add `react-listener-provider`?

### CODE REVIEW

**Comment by jnsdls** [comment]
@aweary you can close #7 as well, since it won't be necessary with this.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-perimeter",
-  "version": "0.2.0",
+  "version": "0.2.1",
```
Comment (aweary) [inline package.json]: Can you undo the version bump? I use `np` to publish new releases

Comment (jnsdls) [reply package.json]: sure

File: src/index.js [hunk 2]
```diff
@@ -59,11 +66,17 @@ export default class Perimeter extends Component {
    */
   componentDidMount() {
     const { handleMouseMove, handleResize, node } = this;
+    const { listener } = this.context;
     if (node) {
       this.bounds = node.getBoundingClientRect();
       this.initialOffset = window.pageYOffset;
-      window.addEventListener('mousemove', handleMouseMove);
-      window.addEventListener('resize', handleResize);
+      if(listener) {
```
Comment (aweary) [inline src/index.js]: Can you add a space before the `if` condition?

File: src/index.js [hunk 3]
```diff
@@ -84,8 +97,14 @@ export default class Perimeter extends Component {
    * if the `once` prop is set to `true`
    */
   removeEventListeners() {
-    window.removeEventListener('mousemove', this.handleMouseMove);
-    window.removeEventListener('resize', this.handleResize);
+    const { listener } = this.context;
+    if(listener){
```
Comment (aweary) [inline src/index.js]: Space before and after the `if` condition please!

File: package.json (hunk 1 above)
Comment (jnsdls) [reply package.json]: done

**Review by aweary — CHANGES_REQUESTED** [review]
This LGTM, I love how simple it is now! Just a couple linting fixes please and this is good to go

**Comment by jnsdls** [comment]
@aweary implemented those changes ;)

**Review by aweary — APPROVED** [review]

**Comment by jnsdls** [comment]
🎉

**Pull request merged**